src/components/expense-documents-input.tsx

```diff
@@ -29,7 +29,7 @@ type Props = {
 
 export function ExpenseDocumentsInput({ documents, updateDocuments }: Props) {
   const [pending, setPending] = useState(false)
-  const { FileInput, openFileDialog, uploadToS3 } = useS3Upload()
+  const { FileInput, openFileDialog, uploadToS3 } = usePresignedUpload() // use presigned uploads to addtionally support providers other than AWS
```

Comment:
Not sure about this (since I'm no expert on js/ts), but you could continue to use the `useS3Upload()` function if `!!process.env.S3_UPLOAD_ENDPOINT`, since it's recommended for various reasons by the `next-s3-upload` lib.

So import both, and set the variables depending on the env var.

If this doesn't work for whatever client/server magic reasons that always happens with next, you may also ignore this comment :)

Reply:
Yeah, that's what I meant about the client/server magic, so then I agree that this is the better solution, but  @scastiel has the last word on the PR anyhow 😄

Reply:
Agree to use `usePresignedUpload` everytime if it doesn’t cause any issue with S3. Plus it’s tricky to use a hook conditionally with React 😉